admin-console/src/main/webapp/src/pages/ApplicationPage.tsx

```diff
@@ -75,42 +96,54 @@ export const ApplicationPage: React.FunctionComponent = () => {
-                                                    id='name' /></FormGroup>
+                                                    defaultValue={ eventType.name }
+                                                    onChange={ handleChange }
+                                                    id='name'
+                                                    name="name"
+                                                /></FormGroup>
                                             <FormGroup label='Display name' fieldId='display-name' isRequired
                                                 helperText='This is the name you want to display on the UI'>
                                                 <TextInput
                                                     type='text'
-                                                    value={ displayName }
-                                                    onChange={ setDisplayName }
+                                                    defaultValue={ eventType.displayName }
+                                                    name="displayName"
+                                                    onChange={ handleChange }
                                                     id='display-name' /></FormGroup>
                                             <FormGroup label='Description' fieldId='description'
                                                 helperText='Optional short description that appears in the UI
                                                 to help admin descide how to notify users.'>
                                                 <TextArea
                                                     type='text'
-                                                    onChange={ setDescription }
-                                                    id='description' /></FormGroup>
+                                                    defaultValue={ eventType.description }
+                                                    onChange={ handleChange }
+                                                    id='description'
+                                                    name="description" /></FormGroup>
                                             <ActionGroup>
-                                                <Button variant='primary' type='submit' isDisabled={ !name || !displayName }
-                                                    onClick= { onSubmit }>Submit</Button>
-                                                <Button variant='link' onClick={ toggle }>Cancel</Button>
+                                                <Button variant='primary' type='button'
+                                                    { ...(newEvent.loading || newEvent.payload?.status !== 200) ?
+                                                        <Spinner /> : eventTypesQuery.payload.value }
+                                                    onClick={ handleSubmit } { ...() => setIsEdit(!isEdit) }>
```

Comment:
What does `{ ...() => setIsEdit(!isEdit) }` do ?

Reply:
i am getting a "... expected " when trying to do this { setIsEdit(!idEdit) } and { ...setIsEdit(!isEdit) } crashes with too many re-renders - though I can just add it to the onClick `onClick={ handleSubmit || setIsEdit(!isEdit }`

Reply:
I actually don't think you need the `() => setIsEdit(!isEdit)` at all.

Reply:
i dont! I have the isEdit on the next line - overthinking what I was trying to do there .. felt I needed to set it more than once for some reason. removed!